cli/module_registry.go

```diff
@@ -621,7 +623,99 @@ func isValidOrgID(str string) bool {
 	return err == nil
 }
 
-func loadManifest(manifestPath string) (moduleManifest, error) {
+type overridePlatform struct {
+	os   string
+	arch string
+	// careful: dev=false doesn't mean non-dev, it means 'don't check dev'. This is not a tri-state.
+	dev bool
+}
+
+// parse one of these from the map key.
+func parseOverridePlatform(raw string) (overridePlatform, error) {
+	if raw == "dev" {
+		return overridePlatform{dev: true}, nil
+	}
+	if len(raw) == 0 {
+		return overridePlatform{}, errors.New("can't parse empty platform key in overrides")
+	}
+	before, after, _ := strings.Cut(raw, "/")
+	if len(before) == 0 || len(after) == 0 {
+		return overridePlatform{}, fmt.Errorf("cannot parse platform key '%s' in overrides", raw)
+	}
+	return overridePlatform{os: before, arch: after}, nil
+}
+
+// returns false if the arguments fall outside op's constraints.
+func (op overridePlatform) check(os, arch string, dev bool) bool {
+	if op.os != "" && op.os != "any" && op.os != os {
+		return false
+	}
+	if op.arch != "" && op.arch != "any" && op.arch != arch {
+		return false
+	}
+	if op.dev && !dev {
+		// note: this isn't a tri-state. op.dev = false means 'don't check', not 'non-dev environment'.
```

Comment:
outdated comment?

Dev means dev environment now, right?

Reply:
good question -- I didn't think carefully about how your tagging change affects the application of tags

the current behavior is to apply every override section that matches. so if someone had a `linux/any` and a `dev`, the linux/any would apply even in dev mode. the override sections would get applied in order and where there's a conflict, the latest would rule. (assuming map order is consistent).

if we change, will someone with a linux override say to themselves 'I'm on linux why isn't my linux override running'?

Reply:
> if someone had a linux/any and a dev, the linux/any would apply even in dev mode.

I think this makes sense if the dev overrides are applied after.

Consider if macOS had a different entrypoint, even if I changed the build step in `dev`, I would want the mac overrides to apply (to fix entrypoint) and then I would also want the dev overrides to stack on top to change the build / makefile / etc. 


 This is a product decision and you are better at those than I am

Reply:
I think if we decide the correct order to apply the changes (sort linux -> linux/arm -> dev) then we would get intuitive behavior.

Reply:
hmm map order is explicitly not preserved in json parsing https://github.com/golang/go/issues/27179

and implicitly seems to be alphabetical, which would give darwin -> dev -> linux :grimacing: 

I think this is going to be a bit surprising no matter what, but your suggestion makes sense -- I'm going to do that + doc in the jsonschema

Reply:
done in 595dafb2b